edx_django_utils/monitoring/internal/backends.py

```diff
@@ -0,0 +1,189 @@
+        span or it might the root span of the process, depending on
+        the backend.
+        """
+
+    @abstractmethod
+    def record_exception(self):
+        """
+        Record the exception that is currently being handled.
+        """
+
+
+class NewRelicBackend(TelemetryBackend):
+    """
+    Send telemetry to New Relic.
+
+    https://docs.newrelic.com/docs/apm/agents/python-agent/python-agent-api/guide-using-python-agent-api/
+    """
+    def __init__(self):
+        if newrelic is None:
+            raise Exception("Could not load New Relic monitoring backend; package not present.")
+
+    def set_attribute(self, key, value):
+        # Sets attribute on the transaction, rather than the current
+        # span, matching historical behavior. There is also an
+        # `add_custom_span_attribute` that would better match
+        # OpenTelemetry's behavior, which we could try exposing
+        # through a new, more specific TelemetryBackend method.
+        #
+        # TODO: Update to newer name `add_custom_attribute`
+        # https://docs.newrelic.com/docs/apm/agents/python-agent/python-agent-api/addcustomparameter-python-agent-api/
+        newrelic.agent.add_custom_parameter(key, value)
+
+    def record_exception(self):
+        # TODO: Replace with newrelic.agent.notice_error()
+        # https://docs.newrelic.com/docs/apm/agents/python-agent/python-agent-api/recordexception-python-agent-api/
+        newrelic.agent.record_exception()
+
+
+class OpenTelemetryBackend(TelemetryBackend):
+    """
+    Send telemetry via OpenTelemetry.
+
+    API reference: https://opentelemetry-python.readthedocs.io/en/latest/
+
+    Call ``OpenTelemetryBackend.initialize()`` from manage.py or gunicorn post-fork
+    hook.
+    """
+    initialized = False
+    init_lock = threading.Lock()
+
+    @classmethod
+    def initialize(cls):
+        """
+        Initialize and configure OpenTelemetry manually.
+
+        This is needed if not using opentelemetry-instrument server
+        startup wrapper (which is incompatible with gunicorn).
+        """
+        # Thread-safe one-time init to support gunicorn's forking model.
+        # (Otherwise we get warnings about re-initializing OTel.)
+        with cls.init_lock:
+            if cls.initialized:
+                return
+            else:
+                cls.initialized = True
+
+        resource = Resource(attributes={
+            SERVICE_NAME: "edxapp-lms"
+        })
+
+        provider = TracerProvider(resource=resource)
+        provider.add_span_processor(BatchSpanProcessor(
+            OTLPSpanExporter(endpoint="http://localhost:4317")
+        ))
+        trace.set_tracer_provider(provider)
+
+        reader = PeriodicExportingMetricReader(OTLPMetricExporter(endpoint="http://localhost:4317"))
+        meterProvider = MeterProvider(resource=resource, metric_readers=[reader])
+        metrics.set_meter_provider(meterProvider)
+
+        DjangoInstrumentor().instrument()
```

Comment:
TODO: I think we instead want to call an auto-instrument function of some kind that will detect all of the available and relevant integrations.